### PR TITLE
add models module, use it

### PR DESCRIPTION
**Pull request by Mattwmaster58**
This PR add `models.py` and uses it. Not every eligible complex type was considered for this module, as this PR aims only to establish it's existence and valid use cases.

### CODE REVIEW

File: pyppeteer/browser_fetcher.py [hunk 1]
```diff
@@ -44,10 +44,10 @@ class RevisionInfo(TypedDict):
...
 
 
-class BrowserOptions(TypedDict, total=False):
-    platform: Platforms
-    path: Path
-    host: str
+# class BrowserFetcherOptions(TypedDict, total=False):
+#     platform: Platforms
+#     path: Path
+#     host: str
```
Comment (Granitosaurus) [inline pyppeteer/browser_fetcher.py]:
```suggestion
```
Or is reserved for future?

**Comment by Granitosaurus** [comment]
Looks good! We should probably include some `Literals` there too, right? There are many options that have are essentially choices. I just commited `WaitTargets` in `lifecycle_watcher.py` for example.

File: pyppeteer/browser_fetcher.py (hunk 1 above)
Comment (Mattwmaster58) [reply pyppeteer/browser_fetcher.py]: Nope, doesn't make sense to use it in the future. All places where this occurs have been expanded to regular args.

**Comment by Mattwmaster58** [comment]
> 
> 
> Looks good! We should probably include some `Literals` there too, right? There are many options that have are essentially choices. I just commited `WaitTargets` in `lifecycle_watcher.py` for example.

For sure. Mouse buttons (`Literal['left', 'right', 'middle']`) are also a good candidate. I say we merge this and add those on the main branch as they are usually very small.

**Pull request merged**